app/domain/authentication/security/extract_request_remote_addr_info.rb

```diff
@@ -0,0 +1,51 @@
+require 'command_class'
+
+module Authentication
+  module Security
+    ExtractRequestRemoteAddrInfo = CommandClass.new(
+      dependencies: {
+        # InputValidation
+      },
+      inputs: [:request]
+    ) do
+
+      extend Forwardable
+      def_delegators :@request, :client_ip, :headers
+
+      def call
+        extract_request_remote_addr_info
+      end
+
+      private
+
+      X_FORWARDED_FOR_HEADER_NAME = 'X-Forwarded-For'
+
+      def extract_request_remote_addr_info
+        xff_header = headers[X_FORWARDED_FOR_HEADER_NAME]
+        xff_header_ip_list = (nil_or_empty xff_header) ? [] : xff_header.to_s.split(/, /)
+        xff_ip_list_length = xff_header_ip_list.length
+
+        if xff_header_ip_list.empty?
+          RequestRemoteAddrInfo.new(
+            client_ip,
+            []
+          )
+        elsif xff_ip_list_length == 1
+          RequestRemoteAddrInfo.new(
+            xff_header_ip_list.first,
+            [client_ip]
+          )
+        elsif xff_ip_list_length > 1
+          RequestRemoteAddrInfo.new(
+            xff_header_ip_list.shift,
+            xff_header_ip_list.uniq
```

Comment:
You should add `remote_addr` to the proxy list , please go over again on the design